bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/UID.java

```diff
@@ -12,129 +12,83 @@
      */
     public UID(String uid) {
-        this(splitToSegments(uid));
-    }
-
-    private static String[] splitToSegments(String uid) {
-        if (uid == null) {
-            throw new IllegalArgumentException("Given uid must not be null.");
-        }
-        return uid.split(SEPARATOR);
+        super(uid);
     }
 
     /**
      * Creates a UID for list of segments.
      *
-     * @param segments
-     *            segments (must not be null)
+     * @param segments segments (must not be null)
      */
     public UID(String... segments) {
-        if (segments == null) {
-            throw new IllegalArgumentException("Given segments argument must not be null.");
-        }
-        int numberOfSegments = getMinimalNumberOfSegments();
-        if (segments.length < numberOfSegments) {
-            throw new IllegalArgumentException("UID must have at least " + numberOfSegments + " segments.");
-        }
-        for (int i = 0; i < segments.length; i++) {
-            String segment = segments[i];
-            validateSegment(segment, i, segments.length);
-        }
-        this.segments = segments;
+        super(segments);
     }
 
     /**
-     * Specifies how many segments the UID has to have at least.
+     * Returns the binding id.
      *
-     * @return
+     * @return binding id
      */
-    protected abstract int getMinimalNumberOfSegments();
-
-    protected String[] getSegments() {
-        return this.segments;
-    }
-
-    protected String getSegment(int segment) {
-        return this.segments[segment];
+    public String getBindingId() {
+        return getSegment(0);
     }
 
-    protected void validateSegment(String segment, int index, int length) {
-        if (!segment.matches(SEGMENT_PATTERN)) {
-            throw new IllegalArgumentException("UID segment '" + segment
-                    + "' contains invalid characters. Each segment of the UID must match the pattern [A-Za-z0-9_-]*.");
-        }
+    protected String[] getSegments() {
+        final List<String> segments = super.getAllSegments();
+        return segments.toArray(new String[segments.size()]);
     }
 
-    /**
-     * Returns the binding id.
-     *
-     * @return binding id
-     */
-    public String getBindingId() {
-        return segments[0];
+    @Override
+    public boolean equals(@Nullable Object other) {
+        return super.equals(other);
     }
 
     @Override
     public String toString() {
-        return getAsString();
+        return super.toString();
     }
 
+    @Override
     public String getAsString() {
-        return Arrays.stream(segments).collect(Collectors.joining(SEPARATOR));
+        return super.getAsString();
     }
 
     @Override
     public int hashCode() {
-        final int prime = 31;
-        int result = 1;
-        result = prime * result + Arrays.hashCode(segments);
-        return result;
-    }
-
-    @Override
-    public boolean equals(Object obj) {
```

Comment:
out of curiosity: why are you able to remove `equals` here but leave `hashCode` with a call to super?

Reply:
Never mind, I just saw the new `equals` implementation...

Reply:
If a bundle uses the `UID` class and import `org.eclipse.smarthome.core.thing` but does not import `org.eclipse.smarthome.core.common` the IDE complains about e.g.:

```text
Access restriction: The method 'GenericUID.hashCode()' is not API (restriction on required project 'org.eclipse.smarthome.core')
```

That has been the reason I added that stubs at the moment.

Reply:
thanks for the explanation.